routes/usersRouter.js

```diff
@@ -0,0 +1,87 @@
+require('dotenv').config()
+const jwt = require('jsonwebtoken')
+const {JWT_SECRET = "neverTell"} = process.env
+const express = require('express');
+const usersRouter = express.Router();
+const { requireUser } = require('./utils');
+
+const {
+  getUserByUsername,
+  getUser,
+  getAllUsers,
+} = require('../db/users');
+
+const {
+	createUser,
+} = require('../db/index');
+
+usersRouter.get('/', async(req, res, next) => {
+    try {
+        const allUsers = await getAllUsers()
+        res.send(allUsers)
+    } catch (error) {
+        next(error)
+    }
+});
+
+usersRouter.post('/register', async (req, res, next) => {
+    const {username, password} = req.body;
+    try {
+        const checkUser = await getUserByUsername(username);
+        if (checkUser) {
+            throw new Error ('A user by that username already exists.')
+        }
+        if (password.length < 8) {
+            throw new Error ('Passwords must be at least 8 characters long')
+        }
+
+        const user = await createUser(req.body)
```

Comment:
Pass in all the db fields explicitly, pulled out from the body above.  
```js
// top of route
const {username, password, firstName, lastName} = req.body;
```
And then when creating
```js
const user = await createUser({username, password, firstName, lastName, /*etc*/})
```